mongoengine/connection.py

```diff
@@ -123,65 +128,89 @@ def disconnect(alias=DEFAULT_CONNECTION_NAME):
 
 def get_connection(alias=DEFAULT_CONNECTION_NAME, reconnect=False):
     global _connections
+
     # Connect to the database if not already connected
     if reconnect:
         disconnect(alias)
 
-    if alias not in _connections:
-        if alias not in _connection_settings:
-            msg = 'Connection with alias "%s" has not been defined' % alias
-            if alias == DEFAULT_CONNECTION_NAME:
-                msg = 'You have not defined a default connection'
-            raise ConnectionError(msg)
-        conn_settings = _connection_settings[alias].copy()
-
-        conn_settings.pop('name', None)
-        conn_settings.pop('username', None)
-        conn_settings.pop('password', None)
-        conn_settings.pop('authentication_source', None)
-        conn_settings.pop('authentication_mechanism', None)
-
-        is_mock = conn_settings.pop('is_mock', None)
-        if is_mock:
-            # Use MongoClient from mongomock
-            try:
-                import mongomock
-            except ImportError:
-                raise RuntimeError('You need mongomock installed '
-                                   'to mock MongoEngine.')
-            connection_class = mongomock.MongoClient
-        else:
-            # Use MongoClient from pymongo
-            connection_class = MongoClient
-
-        if 'replicaSet' in conn_settings:
-            # Discard port since it can't be used on MongoReplicaSetClient
-            conn_settings.pop('port', None)
-            # Discard replicaSet if not base string
-            if not isinstance(conn_settings['replicaSet'], basestring):
-                conn_settings.pop('replicaSet', None)
-            if not IS_PYMONGO_3:
-                connection_class = MongoReplicaSetClient
-                conn_settings['hosts_or_uri'] = conn_settings.pop('host', None)
-
+    # If the requested alias already exists in the _connections list, return
+    # it immediately.
+    if alias in _connections:
+        return _connections[alias]
+
+    # Validate that the requested alias exists in the _connection_settings.
+    # Raise MongoEngineConnectionError if it doesn't.
+    if alias not in _connection_settings:
+        msg = 'Connection with alias "%s" has not been defined' % alias
+        if alias == DEFAULT_CONNECTION_NAME:
+            msg = 'You have not defined a default connection'
+        raise MongoEngineConnectionError(msg)
+
+    def _clean_settings(settings_dict):
+        irrelevant_fields = (
+            'name', 'username', 'password', 'authentication_source',
+            'authentication_mechanism'
+        )
+        return dict(
```

Comment:
Dict comprehension won't work in Py26. Looking forward to the (hopefully near) future when we drop its support :)